jcl/src/java.base/share/classes/openj9/internal/criu/security/CRIUConfigurator.java

```diff
@@ -52,6 +60,51 @@ public final class CRIUConfigurator {
 	public static void setCRIUSecMode(Properties props) {
 		systemProps = props;
 
+		// Get all configured security providers and save all unique services and their associated algorithms
+		// into a cache of type Map. When cache has been populated remove all the providers we loaded.
+		Provider[] providerList = Security.getProviders();
+		for (Provider provider: providerList) {
+			if (debug) {
+				System.out.format("Name: %s%nInformation:%n%s%n", provider.getName(), provider.getInfo());
+			}
+			// If the SunPKCS11 provider is configured we do not want to populate an algorithm cache. The
+			// reason being that the algorithms associated with the SunPKCS11 provider are based upon the
+			// reported mechanisms available in the systems PKCS11 interface. For CRIU we will avoid caching
+			// all the various algorithms and their services since the cache will not be accurate on a
+			// CRIU restore on a different system where we have potentially different PKCS11 mechanisms configured.
+			if (provider.getName().equalsIgnoreCase("SunPKCS11")) {
+				if (provider.isConfigured()) {
+					if (debug) {
+						System.out.println("SunPKCS11 provider is configured. Disable algorithm caching.");
+					}
+					cachedAlgorithms.clear();
+					break;
+				} else {
+					if (debug) {
+						System.out.println("SunPKCS11 provider is not configured. Allow algorithm caching.");
+					}
+				}
+			}
+			Set<Service> serviceList = provider.getServices();
+			for (Service service: serviceList) {
```

Comment:
Please insert a space before `:` and squash to one commit.

Reply:
Made this update to make the line look as so `for (Service service : serviceList)` I then squashed all commits into one commit.